ManalHamdi/assignment2/Assignment2_q2.cc

```diff
@@ -0,0 +1,101 @@
+#include <iostream>
+#include <queue>
+#include <vector>
+
+struct Node {
+	int data;
+	Node* parent;
+	Node* left_child;
+	Node* right_child; 
+};
+class Tree {
+public:
+	Node* root;
+	Tree() {
+		root = nullptr;
+	}
+	Node* CreateNode(int data) {
+		Node* temp = new Node;
+		temp->data = data;
+		temp->left_child = nullptr;
+		temp->right_child = nullptr;
+		temp->parent = nullptr;
+		return temp;
+	} 
+	void AppendTree(Node* parent, Node* node_to_append, bool append_to_left) {
+		node_to_append->parent = parent;
+		if (append_to_left) {
+			parent->left_child = node_to_append;
+		}
+		else
+			parent->right_child = node_to_append;
+	}
+	void CreateRoot(int data) {
```

Comment:
You can do this in the constructor(this is the preferred way of initalising objects):

```
class Tree {
    public:
         Tree(int data) {
              root = CreateNode(data);
              // You don't need to set parent to nullptr because you're already doing it in CreateNode.
         }
}
```

Then, in your main:

```
Tree tree(1); // creates a tree with root '1';

Node* node2 = ...
```